src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -62,24 +65,46 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
                 {
                     // Process bypass list for manual setting.
                     string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>(list.Length);
 
                     foreach (string value in list)
                     {
                         string tmp = value.Trim();
+                        if (tmp.Length == 0)
+                        {
+                            continue;
+                        }
                         if (tmp == "<local>")
                         {
                             _bypassLocal = true;
                             continue;
                         }
-                        if (tmp.Length > 0)
+                        try
+                        {
+                            // Escape any special characters and unescape * to get wildcard pattern match.
+                            Regex re = new Regex(Regex.Escape(tmp).Replace("\\*", ".*?") + "$",
+                                            RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+                            _bypass.Add(re);
+                        }
+                        catch
                         {
-                            tmpList.Add(tmp);
+                            if (NetEventSource.IsEnabled)
+                            {
+                                NetEventSource.Info(this, "Failed to process " + tmp + " from bypass list.");
+                            }
                         }
                     }
-                    if (tmpList.Count > 0)
+                }
+                if (_bypassLocal)
+                {
+                    _localIp =  new List<IPAddress>();
+                    foreach (NetworkInterface netInterface in NetworkInterface.GetAllNetworkInterfaces())
                     {
-                        _bypass = tmpList.ToArray();
+                        IPInterfaceProperties ipProps = netInterface.GetIPProperties();
+                        foreach (UnicastIPAddressInformation addr in ipProps.UnicastAddresses)
+                        {
+                            _localIp.Add(addr.Address);
+                        }
```

Comment:
So my earlier comment about allocations... those were nothing compared to the amount of allocation and work happening here :)  But nothing to be addressed right now.